cookbooks/bcpc-hadoop/libraries/utils.rb

```diff
@@ -376,44 +335,41 @@ def process_restarted_after_failure?(restart_failure_time, process_identifier)
   require 'time'
   begin
     start_time = process_start_time(process_identifier)
-    if start_time.nil?
-      return false
-    elsif Time.parse(restart_failure_time).to_i < Time.parse(start_time).to_i
-      Chef::Log.info ("#{process_identifier} seem to be started at #{start_time} after last restart failure at #{restart_failure_time}")
+    if start_time? && (Time.parse(restart_failure_time).to_i < Time.parse(start_time).to_i)
+      Chef::Log.info("#{process_identifier} seem to be started at #{start_time} after last restart failure at #{restart_failure_time}")
       return true
-    else
-      return false
     end
+
+    return false
   end
 end
 
 def user_exists?(user_name)
   user_found = false
   chk_usr_cmd = "getent passwd #{user_name}"
   Chef::Log.debug("Executing command: #{chk_usr_cmd}")
-  cmd = Mixlib::ShellOut.new(chk_usr_cmd, :timeout => 10).run_command
-  if cmd.exitstatus == 0
-    user_found = true
-  end
-  return user_found
+  cmd = Mixlib::ShellOut.new(chk_usr_cmd, 'timeout' => 10).run_command
+  user_found = true if cmd.exitstatus.zero?
+
+  user_found
 end
 
 def group_exists?(group_name)
   chk_grp_cmd = "getent group #{group_name}"
   Chef::Log.debug("Executing command: #{chk_grp_cmd}")
-  cmd = Mixlib::ShellOut.new(chk_grp_cmd, :timeout => 10).run_command
-  return cmd.exitstatus == 0 ? true : false
+  cmd = Mixlib::ShellOut.new(chk_grp_cmd, 'timeout' => 10).run_command
+  cmd.exitstatus.zero ? true : false
 end
 
 def get_group_action(group_name)
-  return group_exists?(group_name) ? :manage : :create
+  group_exists?(group_name) ? :manage : :create
 end
 
-def has_vip?
+def vip?
   cmd = Mixlib::ShellOut.new(
-    "ip addr show", :timeout => 10
+    'ip addr show', 'timeout' => 10
```

Comment:
Can you please explain a little bit more?

Reply:
Looks like it's using `Mixlib::ShellOut` to me...

However, rubocop will yell at you for not using Ruby 1.9+ hash syntax, so rather than:
`'timeout' => 10`
it should probably read:
`timeout: 10`

Reply:
Sorry, somehow I completely misread the above code. Please disregard.